pkg/volume/csi/csi_test.go

```diff
@@ -306,6 +307,7 @@ func TestCSI_VolumeAll(t *testing.T) {
 				// creates VolumeAttachment and blocks until it is marked attached (done by external attacher)
 				go func(spec *volume.Spec, nodeName types.NodeName) {
 					attachID, err := volAttacher.Attach(spec, nodeName)
+					close(waitForAttachDone)
```

Comment:
Do we want to close before checking error?

Reply:
yes, we don't want to hang the test in case of an error

Reply:
Ah yes my confusion was around thinking this was in the main thread of execution here, which would mean that if the encapsulating function returned first that the routine would be terminated. Not sure what the odds of that even happening would be, but this goroutine is not encapsulated by the main thread of execution. However, let's say the last test case was being run and we hit an error here, could we feasibly exit the encapsulating function _and_ the main thread of execution prior to calling `t.Errorf`, thus causing the test to pass (assuming no other errors)?